### ISSUE TITLE
Formalise Benchmarks

### ISSUE THREAD

**Issue by pedromxavier** [JuliaPy/PythonCall.jl#308]
## Rationale

Create a formal benchmark pipeline to compare
- Python
- PythonCall (dev)
- PythonCall (stable)
- PyCall

## Resources
- [BENCHMARKS.md](https://github.com/cjdoris/PythonCall.jl/blob/main/BENCHMARKS.md)

## References
1. [BenchmarkTools.jl](https://juliaci.github.io/BenchmarkTools.jl/stable/)
2. [pyperf](https://pyperf.readthedocs.io/en/latest/)

_Originally posted by @cjdoris in https://github.com/cjdoris/PythonCall.jl/issues/300#issuecomment-1547350528_

**Comment by LilithHafner** [comment]
IMO this is still relevant, it should be re-opened and added to a milestone so that it is not automatically re-closed as stale.

**Comment by cjdoris** [comment]
Indeed, I like this PR, just haven't had a chance to properly review it.

**Comment by pedromxavier** [comment]
I had a similar task in another project and some of the ideas converged to slightly different approaches. I will be happy to update this PR soon, probably during the next weekend.

**Comment by cjdoris** [comment]
Sounds good!

**Comment by pedromxavier** [comment]
Since PythonCall is not v1 yet, we have to decide on how we want to compare the different branches under interface changes. Are we going to keep separate suites for dev and stable or not?
